minject/registry.py

```diff
@@ -248,21 +324,95 @@ def get(
             return _unwrap(self._by_name.get(key, RegistryWrapper(cast(T, default))))
 
         meta = _get_meta_from_key(key)
+        maybe_class = self._get_class_if_registered(key, meta)
+        if maybe_class is not None:
+            return maybe_class
+
+        return _unwrap(self._get_by_metadata(meta, default))
+
+    async def _aget(self, key: "RegistryKey[T]") -> Optional[T]:
+        if not self._async_can_proceed:
+            raise RegistryAPIError("cannot use aget outside of async context")
+
+        if isinstance(key, str):
+            raise RegistryAPIError("cannot use aget with string keys. Use get instead.")
+
+        meta = _get_meta_from_key(key)
+        maybe_class = self._get_class_if_registered(key, meta)
+        if maybe_class is not None:
+            return maybe_class
+
+        by_meta = await self._aget_by_metadata(meta)
+        return _unwrap(by_meta)
+
+    def _get_class_if_registered(
```

Comment:
NOTE: THIS METHOD IS THE ONE I AM MOST NERVOUS ABOUT IN THE PR

I added `_get_class_if_registered` to try and remove duplicated code in `get()`, `_aget()`, and `aget()` around checking if a class was already created in the `registry`.

This is the trickiest change to an existing implementation in this PR, and should probably be given significant attention around correctness in a review.